### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by TadesseDev**
# Changes introduced. 😎
---
- Add RSpec as a gem ➕
- Create a Request spec file for all my controllers ✍🏼
> - Test If response status was correct ✔
> - Test If a correct template was rendered.✔
> - Test If the response body includes correct placeholder text.✔

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    it 'response is ok for show' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'render show template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'users/user_id contains place holder text' do
+      expect(response.body).to match('this is a user with ID')
+    end
```
Comment (alexander16108) [inline spec/requests/users_spec.rb]: Good job on the test cases, however, but your Spec file should include tests that ensure the response body includes the correct placeholder text.
Example :` expect(response.body).to include("Welcome to user index page")`

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    it 'render the right page' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'user/posts containes place holder text' do
+      expect(response.body).to match('this is a post index')
+    end
+  end
+  describe 'posts show action tests' do
+    before(:example) { get('/users/:user_id/posts/:id') }
```
Comment (alexander16108) [inline spec/requests/posts_spec.rb:18]: Good job on the test cases, however, but your Spec file should include tests that ensure the response body includes the correct placeholder text.
Example : `expect(response.body).to include("Welcome to user index page")`

>

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Tadesse-Alemayehu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by TadesseDev** [comment]
@alexander16108 Thanks for the review. I have user https://github.com/Tadesse-Alemayehu/Rail-blog-app/blob/c4c6bf13753356293d0fda76919555c3baab8193/spec/requests/users_spec.rb#L26-L28 to test  the response body includes the correct placeholder text. 

But ill make sure to change it if it's required that way.

**Review by Deepakdanger — APPROVED** [review]
Hi @Tadesse-Alemayehu, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TadesseDev** [comment]
Thanks, @Deepakdanger. Nice to have your review. 🙏🏼

**Pull request merged**